api/construction_service.go

```diff
@@ -114,21 +114,33 @@ func (s *Server) ConstructionMetadata(ctx context.Context, r *types.Construction
 		return nil, xerr
 	}
 	height := uint64(0)
+	client := s.ConstructionAccessNodes.Client()
+	hdr, err := client.LatestBlockHeader(ctx)
+	if err != nil {
+		return nil, wrapErr(errInternal, err)
+	}
 	if opts.Inner {
 		if opts.SequenceNumber == -1 {
-			// TODO(tav): Perhaps look up the Proxy Vault's nonce value from the
-			// chain to make testing easier.
-			return nil, wrapErrorf(
-				errInvalidConstructOptions,
-				"sequence_number not specified in previous call to /construction/preprocess",
+			resp, err := client.Execute(
+				ctx, hdr.Id, s.scriptGetProxyNonce,
+				[]cadence.Value{cadence.BytesToAddress(opts.Payer)},
 			)
+			if err != nil {
+				return nil, wrapErrorf(
+					errFailedAccessAPICall,
+					"failed to execute get_proxy_nonce: %s", err,
+				)
+			}
+			nonce, ok := resp.ToGoValue().(int64)
```

Comment:
If only proxy account has deposit transfer, will it still come back the default nonce -1?

Reply:
In `GetProxyNonce` within `script/script.go`, we `return vault.lastNonce + 1`. So, for the very first "inner transaction" construction, this will return a `nonce` value of `0`.

Does that explain it? Sorry if I didn't understand your question fully.

Reply:
If there are two deposits into proxy account, then proxy nonce is still 0. The deposit, the contract does not update nonce.  My question was to check the balance and nonce for proxy account.